plugin/al/lib/AL_USDMaya/AL/usdmaya/nodes/Transform.cpp

```diff
@@ -454,12 +461,65 @@ MStatus Transform::validateAndSetValue(
             }
             transform()->setPrim(UsdPrim(), this);
         }
+        dirtyMatrix();
         return MS::kSuccess;
     }
 
     return MPxTransform::validateAndSetValue(plug, handle, context);
 }
 
+//----------------------------------------------------------------------------------------------------------------------
+MStatus Transform::preEvaluation(const MDGContext& context, const MEvaluationNode& evaluationNode)
+{
+    MDGContextGuard contextGuard(context);
+    if (!context.isNormal()) {
+
+        MDataBlock dataBlock = forceCache(*(MDGContext*)&context);
+        auto*      data = inputDataValue<MayaUsdStageData>(dataBlock, m_inStageData);
+        if (data && data->stage) {
+            MString path = inputStringValue(dataBlock, m_primPath);
+            SdfPath primPath;
+            UsdPrim usdPrim;
+            if (path.length()) {
+                primPath = SdfPath(path.asChar());
+                usdPrim = data->stage->GetPrimAtPath(primPath);
+            }
+            getTransMatrix()->setPrim(usdPrim, this);
+        }
+
+        MTime time;
+        context.getTime(time);
+        MTime theTime = (time - inputTimeValue(dataBlock, m_timeOffset))
+            * inputDoubleValue(dataBlock, m_timeScalar);
+        UsdTimeCode usdTime(theTime.as(MTime::uiUnit()));
+
+        getTransMatrix()->updateToTime(usdTime);
+        MHWRender::MRenderer::setGeometryDrawDirty(thisMObject());
```

Comment:
Notifying renderers about changes in a non-normal context is not allowed. Why this code is necessary?

Reply:
The context of this change was an issue we had with evaluation caching, it dates back to late 2020.
Let me dig into this further.

Reply:
I can't answer why it was written that way but I can confirm evaluation caching works properly without it (in fact, it's not needed in post evaluation either).

Reply:
Let me update this internally first, then I'll apply the changes here.
In the meantime, any help to solve the remaining failing test would be appreciated.

Reply:
@kxl-adsk , code has been updated.